### PR TITLE
Add allure-python

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

This is intended to replace https://github.com/conda-forge/allure-python-commons-feedstock which produces one package (`allure-python-commons`) while this produces the entire suite of related packages (via `outputs`).

### CODE REVIEW

File: recipes/allure-pytest/meta.yaml
Comment (kenodegard) [inline recipes/allure-pytest/meta.yaml:121]: @nsoranzo as the maintainer of https://github.com/conda-forge/allure-python-commons-feedstock are you still interested in being a maintainer here?

Comment (nsoranzo) [reply recipes/allure-pytest/meta.yaml:121]: allure-python-commons is a dependency of a bioconda package I co-maintain, so happy to be listed here, thanks!

**Comment by kenodegard** [comment]
@nsoranzo you mentioned (https://github.com/conda-forge/allure-python-commons-feedstock/pull/21#issuecomment-1593223066) needing a PR in https://github.com/conda-forge/feedstock-outputs as well.

It looks like much of that repo is auto generated so my suspicion is this will fix itself once this gets merged?

**Comment by kenodegard** [comment]
@conda-forge/help-python, ready for review!

**Comment by nsoranzo** [comment]
> @nsoranzo you mentioned ([conda-forge/allure-python-commons-feedstock#21 (comment)](https://github.com/conda-forge/allure-python-commons-feedstock/pull/21#issuecomment-1593223066)) needing a PR in https://github.com/conda-forge/feedstock-outputs as well.
> 
> It looks like much of that repo is auto generated so my suspicion is this will fix itself once this gets merged?

Not sure if this is taken care of automatically, I suggested it because it's mentioned in the [Renaming Packages guideline](https://conda-forge.org/docs/orga/guidelines.html#renaming-packages): "If an existing feedstock has already created a package of the same name, then you may need to add the new feedstock to the [feedstock-outputs](https://github.com/conda-forge/feedstock-outputs) map."

File: recipes/allure-pytest/meta.yaml
Comment (beckermr) [inline recipes/allure-pytest/meta.yaml:20]: This won't work since noarch is only built on linux. In this case, you can simply rempve the windows stuff.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: recipes/allure-pytest/meta.yaml
Comment (kenodegard) [reply recipes/allure-pytest/meta.yaml]: ahhh, makes sense! yay less code!

**Comment by kenodegard** [comment]
@beckermr thanks for reviewing and merging! whats the next steps for https://github.com/conda-forge/allure-python-commons-feedstock? is this manually archived?

**Comment by beckermr** [comment]
Yes you can. You maybe need to submit a pr to add the output to our maps if it is a duplicate.

**Pull request merged**
